lib/orogen/gen/tasks.rb

```diff
@@ -533,6 +533,20 @@ def generate
                     hidden_operation("__orogen_getTID", "    #ifdef HAS_GETTID\nreturn syscall(SYS_gettid);\n#else\nreturn 0;\n#endif").
                         returns("int").
                         doc("returns the PID for this task")
+
+                    if extended_state_support?
```

Comment:
In this case the functionality would only be available to tooling, that can load the typekit. There are cases, when you don't have typekits locally installed, but want to inspect remote tasks.